<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2016-11-11" />

<title>Making HumanPaulMyeloid package</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="floating-toc.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script src="libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dataPackages</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jhsiao999/dataPackages-site">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Making HumanPaulMyeloid package</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2016-11-11</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#setting-up">Setting up</a></li>
<li><a href="#making-eset">Making eSet</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<div id="setting-up" class="section level3">
<h3>Setting up</h3>
<p>Load packages</p>
<pre class="r"><code>library(data.table)
library(Biobase)</code></pre>
<p>Make directories.</p>
<pre class="r"><code>data_dir &lt;- &quot;../data/HumanPaulMyeloid&quot;</code></pre>
</div>
<div id="making-eset" class="section level3">
<h3>Making eSet</h3>
<p>Format Raw data.</p>
<pre class="r"><code># import read count matrix
expr_data &lt;- read.table(file.path(data_dir, &quot;GSE72857_umitab.txt&quot;),
                        header = TRUE, sep = &quot;\t&quot;, fill = TRUE,
                        stringsAsFactors = FALSE)
counts &lt;- as.matrix(expr_data)
dim(counts)


# import annotation matrix
col_data &lt;- read.table(file.path(data_dir, &quot;GSE72857_experimental_design_copy.txt&quot;),
                        header = TRUE, sep = &quot;\t&quot;, fill = TRUE,
                        stringsAsFactors = FALSE)
rownames(col_data) &lt;- colnames(counts)
  
meta_list &lt;- data.frame(
  labelDescription = c(&quot;well identifier&quot;,
                       &quot;Illumina sequencing batch ID&quot;,
                       &quot;amplification batch ID (of 192 single cells)&quot;,
                       &quot;well coordinates&quot;,
                       &quot;subject mouse ID&quot;, 
                       &quot;sorting batch ID (384 wells plates)&quot;, 
                       &quot;Information of the experiment, sorting scheme, and genetic background of mouse&quot;,
                       &quot;Pool barcode (may overlap between different sorting batches&quot;,
                       &quot;Single cell barcode&quot;,
                       &quot;random molecular tag sequence&quot;,
                       &quot;Number of cells sorted to well (0/1)&quot;,
                       &quot;CD 34 protein levels (if index-sorted)&quot;,
                       &quot;FcgR3 protein levels (if index-sorted)&quot;),
                    row.names = colnames(col_data))

# make featureData
row_data &lt;- data.frame(gene_names = rownames(counts),
                       stringsAsFactors = FALSE)
rownames(row_data) &lt;- rownames(counts)


# putting all together
HumanPaulMyeloid &lt;- new(&quot;ExpressionSet&quot;,
                    exprs = as.matrix(counts),
                    phenoData = new(&quot;AnnotatedDataFrame&quot;,
                                    data = col_data,
                                    varMetadata = meta_list),
                    featureData = new(&quot;AnnotatedDataFrame&quot;,
                                      data = row_data),
                    experimentData = new(&quot;MIAME&quot;,
                                         title = &quot;Amit Paul Myeloid&quot;))</code></pre>
<p>Format the version of the data that was processed in Wishbone.</p>
<pre class="r"><code>expr_normed &lt;- read.csv(file.path(data_dir,&quot;wishbone_mouse_marrow_scrnaseq.csv&quot;),
                   header = TRUE, 
                   fill = TRUE,
                   stringsAsFactors = FALSE)
rownames(expr_normed) &lt;- expr_normed[,1]
expr_normed &lt;- expr_normed[,-1]
expr_normed &lt;- t(expr_normed)
str(expr_normed)

# TBD: reformat the gene names
# currently it&#39;s not in standard Ensembl gene format

# Subset cells used in the normalize data.
is_cell_included &lt;- match(colnames(expr_normed), 
                          pData(HumanPaulMyeloid)$Well_ID, nomatch = 0)
pdata_filter &lt;- pData(HumanPaulMyeloid)[is_cell_included,]
rownames(pdata_filter) &lt;- pdata_filter$Well_ID
str(pdata_filter)

# make featureData
fdata_filter &lt;- data.frame(gene_names = rownames(expr_normed))
rownames(fdata_filter) &lt;- fdata_filter$gene_names
str(fdata_filter)

# putting all together
HumanPaulMyeloid_wishbone_normed &lt;- new(&quot;ExpressionSet&quot;,
                    exprs = as.matrix(expr_normed),
                    phenoData = new(&quot;AnnotatedDataFrame&quot;,
                                    data = pdata_filter,
                                    varMetadata = meta_list),
                    featureData = new(&quot;AnnotatedDataFrame&quot;,
                                      data = fdata_filter),
                    experimentData = new(&quot;MIAME&quot;,
                                         title = &quot;Amit Paul Myeloid - filtered and normlized for Wishbone paper&quot;))</code></pre>
<p>Save all data to the same rda.</p>
<pre class="r"><code>save(HumanPaulMyeloid, HumanPaulMyeloid_wishbone_normed,
        file = &quot;../data/all-rda/HumanPaulMyeloid.rda&quot;)</code></pre>
</div>
<div id="session-information" class="section level3">
<h3>Session information</h3>
<pre class="r"><code>sessionInfo()</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
