<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2016-08-26" />

<title>Making MouseZeiselBrain package</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="floating-toc.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single cell RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Data List</a></li>
        <li><a href="dataDescription.md">Data Description</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jhsiao999/dataPackages-site/site">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Making MouseZeiselBrain package</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2016-08-26</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#load-packages">Load packages</a></li>
<li><a href="#zeisel-et-al.-2015-mouse-brain-umi">Zeisel et al., 2015, Mouse brain + UMI</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<div id="load-packages" class="section level3">
<h3>Load packages</h3>
</div>
<div id="zeisel-et-al.-2015-mouse-brain-umi" class="section level3">
<h3>Zeisel et al., 2015, Mouse brain + UMI</h3>
<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60361" class="uri">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60361</a></p>
<blockquote>
<p>We have applied a recently developed, highly accurate and sensitive single-cell RNA-seq method (STRT/C1) to perform a molecular census of two regions of the mouse cerebral cortex: the somatosensory cortex and hippocampus CA1. We isolated cells fresh from somatosensory cortex (S1) and hippocampus CA1 area of juvenile (P22 - P32) CD1 mice, 33 males and 34 females. Cells were collected without selection, except that 116 cells were obtained by FACS from 5HT3a-BACEGFP transgenic mice. A total of 76 Fluidigm C1 runs were performed, each attempting 96 cell captures and resulting in 3005 high-quality single-cell cDNAs, containing Unique Molecular Identifiers allowing counting of individual mRNA molecules, even after PCR amplification.</p>
</blockquote>
<pre class="r"><code># import annotation matrix, conver to a DatFrame object
df_col_data &lt;- as.matrix(
    fread(&quot;../data/MouseZeiselBrain/expression_mRNA_17-Aug-2014.txt&quot;) )[1:10, ]

col_data &lt;- data.frame(
    tissue = colnames(df_col_data)[-(1:2)],
    group_no = round(as.numeric(df_col_data[1, -(1:2)])),
    total_mrna_vol = as.numeric(df_col_data[2, -(1:2)]),
    well_id = round(as.numeric(df_col_data[3, -(1:2)])),
    sex = factor(as.numeric(df_col_data[4, -(1:2)]),
                 levels = c(-1, 1, 0),
                 labels = c(&quot;male&quot;, &quot;female&quot;, &quot;transgenic&quot;)),
    age = as.numeric(df_col_data[5, -(1:2)]),
    diameter = as.numeric(df_col_data[6, -(1:2)]),
    cell_id = df_col_data[7, -(1:2)],
    level1_class = df_col_data[8, -(1:2)],
    level2_class = df_col_data[9, -(1:2)],
    stringsAsFactors = FALSE)
rownames(col_data) &lt;- col_data$cell_id



# import data
df &lt;- as.matrix(
    read.table(&quot;../data/MouseZeiselBrain/GSE60361_C1-3005-Expression.txt&quot;,
    header = TRUE, quote = &quot;&quot;, sep = &quot;\t&quot;,
    stringsAsFactors = FALSE))

# save counts to a matrix object
counts &lt;- matrix(as.numeric(df[ ,-1]),
                    nrow = NROW(df),
                    byrow = FALSE)
colnames(counts) &lt;- colnames(df)[-1]
rownames(counts) &lt;- df[,1]


# two genes are duplicates...
rownames(counts)[which(duplicated(rownames(counts)))]

# check the difference between the duplicates
table((counts[rownames(counts) == &quot;Mar-02&quot;, ])[1,])
table((counts[rownames(counts) == &quot;Mar-02&quot;, ])[2,])

# check the difference between the duplicates
table((counts[rownames(counts) == &quot;Mar-01&quot;, ])[1,])
table((counts[rownames(counts) == &quot;Mar-01&quot;, ])[2,])

# removing these two genes as I am not sure the duplicates
# are drastically different in gene abundance levels
counts &lt;- counts[which(rownames(counts) != &quot;Mar-01&quot; &amp; rownames(counts) != &quot;Mar-02&quot;), ]

# remote X in the beginning of column names
counts_colnames &lt;- gsub(&quot;X&quot;, replacement = &quot;&quot;, x = colnames(counts))
colnames(counts) &lt;- counts_colnames

# feature data
row_data &lt;- data.frame(gene_name = as.character(rownames(counts)),
                       stringsAsFactors = FALSE)
rownames(row_data) &lt;- rownames(counts)

# match count matrix with annotation data.frame
match_cell_id &lt;- match(col_data$cell_id, colnames(counts))
counts &lt;- counts[ ,match_cell_id]

meta_list &lt;- data.frame(
    labelDescription = c(&quot;Tissue type&quot;,
                         &quot;Group nubmer&quot;,
                         &quot;Total mRNA volume&quot;,
                         &quot;Well ID&quot;,
                         &quot;Sex&quot;,
                         &quot;Age (in days)&quot;,
                         &quot;Cell diameter&quot;,
                         &quot;Cell unique ID&quot;,
                         &quot;Level 1 class&quot;,
                         &quot;Level 2 class&quot;),
    row.names = colnames(col_data) )


# putting all together
MouseZeiselBrain &lt;- new(&quot;ExpressionSet&quot;,
                        exprs = as.matrix(counts),
                        phenoData = new(&quot;AnnotatedDataFrame&quot;,
                                        data = col_data,
                                        varMetadata = meta_list),
                        featureData = new(&quot;AnnotatedDataFrame&quot;,
                                          data = row_data),
                        experimentData = new(&quot;MIAME&quot;,
                                             title = &quot;Zeisel Mouse Brain&quot;))


save(MouseZeiselBrain,
     file = &quot;../data/all-rda/MouseZeiselBrain.rda&quot;)</code></pre>
</div>
<div id="session-information" class="section level3">
<h3>Session information</h3>
<pre class="r"><code>sessionInfo()</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
